Shapemapper

guowei-he opened the pull request:
* [x] I have read the guidelines above.
* [x] This PR adds a new recipe.
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

@bioconda/core Please review as it's my first recipe. A bit unfortunate that the code never returns exit code 0 unless adequate inputs are provided (e.g., there is no "./app -h"). Thank you!

bgruening commented on recipes/shapemapper/meta.yaml:
```diff
@@ -0,0 +1,40 @@
+    home: http://www.chem.unc.edu/rna/software.html
+    license: GPL
+    license_file: COPYING
+    summary: ShapeMapper converts raw sequencing files into mutational profiles, creates SHAPE reactivity plots, and provides extensive diagnostic information useful for experiment analysis and troubleshooting. (From homepage)
```
can you remove the (from homepage)?

bgruening commented on recipes/shapemapper/meta.yaml:
```diff
@@ -9,6 +9,7 @@ source:
 build:
     number: 0
     skip: True # [osx]
+    skip: True  # [not py27]
```
this does not work, you should write something like `# [osx or py3k]`

bgruening commented on recipes/shapemapper/meta.yaml:
```diff
@@ -25,10 +26,11 @@ requirements:
         - matplotlib 
         - httplib2 
-        - rnastructure
+        - rnastructure # [not osx]
```
the skip is here not needed if you skip the entire build.

bgruening commented on recipes/shapemapper/meta.yaml:
```diff
@@ -25,10 +26,11 @@ requirements:
 
 test:
     commands:
+        - type ShapeMapper.py
```
does the python script have a `--help` or similar?

guowei-he commented:
*@bgruening*
Thanks for the review!

   1. does this program has a -v like argument?
      1. C(++): I think the original code tried to do a '-h'. But there
      seems to be a bug when it tries to use 'getopt_long_only'. So it doesn't
      work for all executables.
      2. Python: Can't find anything in the source code. It only check the
      number of arguments.
   2. Can you test the python files?
      1. Added.
   3. Is it python2 and python3?
   1. 2.7 and fixed the skip
   4. (from homepage)
      1. Removed.
   5. Requirements with # [not osx]
      1. Removed # [not osx]


On Sun, Apr 9, 2017 at 2:24 PM, Björn Grüning <notifications@github.com>
wrote:

> *@bgruening* commented on this pull request.
> ------------------------------
>
> In recipes/shapemapper/meta.yaml
> <https://github.com/bioconda/bioconda-recipes/pull/4387#discussion_r110537844>
> :
>
> > @@ -9,6 +9,7 @@ source:
>  build:
>      number: 0
>      skip: True # [osx]
> +    skip: True  # [not py27]
>
> this does not work, you should write something like # [osx or py3k]
> ------------------------------
>
> In recipes/shapemapper/meta.yaml
> <https://github.com/bioconda/bioconda-recipes/pull/4387#discussion_r110537851>
> :
>
> > @@ -25,10 +26,11 @@ requirements:
>          - bowtie2
>          - matplotlib
>          - httplib2
> -        - rnastructure
> +        - rnastructure # [not osx]
>
> the skip is here not needed if you skip the entire build.
> ------------------------------
>
> In recipes/shapemapper/meta.yaml
> <https://github.com/bioconda/bioconda-recipes/pull/4387#discussion_r110537864>
> :
>
> >
>  test:
>      commands:
> +        - type ShapeMapper.py
>
> does the python script have a --help or similar?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/bioconda/bioconda-recipes/pull/4387#pullrequestreview-31720498>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AE4szEYWfFkX923C_XtiUAONjj5A2Xdvks5ruLHJgaJpZM4M39fT>
> .
>



-- 
*Guowei HE*
HPC Application Specialist
NYU Abu Dhabi
Office Tel (UAE): +971 2 628 4698 <+971%202%20628%204698>

New York University - Abu Dhabi (NYUAD)
Saadiyat Campus - A2, Room-132B
P.O. Box 129188
Abu Dhabi, United Arab Emirates

https://wikis.nyu.edu/display/ADRC/HPC
Please consider the environment before printing this email.

bgruening commented:
I suspect that these tests will be not sufficient long term, can you report this problem upstream? So that we get real test in the next version?
Also keep in mind that you can do `./binary --help | grep 'specific error message'` as a test. This might be a better test.

guowei-he commented:
@bgruening 

Indeed. I don't think they are sufficient. Unfortunately, I don't think they have public git for it.
Thanks for the grep idea! Let me do this.
